### ISSUE TITLE
Add CProxy

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/virtualc64#241]
Problem: 

- It doesn't really make sense to implement new features in Objective-C (such as TouchBar support). 
- Therefore, we finally want to implement new features in Swift
- Before we can add any Swift, we need to get rid of all direct calls to C++ from the GUI
- Problem: ObjCProxy is not complete, so we have two choices: 
   (1) Add missing stuff to ObjCProxy
   (2) Replace ObjCProxy by a CProxy

We go for (2) because of two reasons:
- It's pretty easy to call C code from swift 
- We are able to replace all ObjC code (eventually)

**Comment by dirkwhoffmann** [comment]
Things to to: 

1. Completion C-API
2. Remove ObjProxy calls
3. Remove ObjProxy object (in XIB file), manually create C64 object in c64_new()
4 Remove any C++ related stuff from ObjectiveC code

**Comment by dirkwhoffmann** [comment]
Problem: We can't get rid of the ObjC proxy. Otherwise, undo can't be implemented.

New plan: 
1. Complete ObjectiveC proxy
2. Move all enums to C compatible header files (VIC_defs.h, SID_defs.h) etc. 
3. Remove "#include "C64.h" from all Objective-C file except Proxy (this will eliminate C++ dependencies)

**Comment by dirkwhoffmann** [comment]
Canceled
